app/docs/md/learn/practices/testing.md

```diff
@@ -85,3 +85,40 @@ test('Shut down local server', async t => {
 })
 ```
 
+## Component testing
+
+If you'd like to run functional component tests, we recommend using [WebdriverIO](https://webdriver.io/) as a browser automation framework. The benefits of using an actual browser compared to JSDOM outweigh the minimal speed detriment that comes with spinning up a browser instance. Furthermore, WebdriverIO is based on [WebDriver](https://www.w3.org/TR/webdriver/), a browser automation web standard which guarantees that your tests interact with your component as closely as possible to how a user would.
+
+To add a WebdriverIO test harness to your project, run:
+
+```shell
+npm init wdio@latest ./
+```
+
+The command initializes a configuration wizard that helps you set-up the test harness. Make sure to select on the first question: "_Where should your tests be launched?_" the answer: "_browser - for unit and component testing in the browser_".
+
+An example test file should be created. Replace the content with e.g.:
+
+```javascript
+import { expect, browser, $$ } from '@wdio/globals'
+import enhance from '@enhance/ssr'
+
+// see actual component here: https://github.com/webdriverio/component-testing-examples/blob/main/enhance/app/elements/my-header.mjs
+import MyHeader from '../../app/elements/my-header.mjs'
+
+describe('Enhance Framework', () => {
+  it('should render MyHeader element correctly', async () => {
+    const html = enhance({
+      elements: {
+        'my-header': MyHeader
+      }
+    })
+    const actual = document.createElement('div')
+    actual.innerHTML = (html`<my-header></my-header>`).replace(/<(\/*)(html|head|body)>/g, '')
```

Comment:
Some tweaks to the regex:
- The first capture group is unnecessary
- The quantifier should allow for a single occurrence of `/`, rather than any number of them
- Whitespace after the `/` should be supported. 

```suggestion
    actual.innerHTML = (html`<my-header></my-header>`).replace(/<\/?\s*(html|head|body)>/g, '')
```